lib/land/trackers/api_tracker.rb

```diff
@@ -206,24 +227,73 @@ def user_agent
                      Land.config.blank_user_agent_string
 
         @user_agent = UserAgent[user_agent]
+        @user_agent.user_agent_type = UserAgentType['user']
 
         if Land.config.identify_crawlers && defined?(CrawlerDetect)
           crawler_detect = CrawlerDetect.new(user_agent)
-          user_agent_type = crawler_detect.is_crawler? ? 'crawl' : 'api'
-          @user_agent.update(user_agent_type: UserAgentType[user_agent_type])
+          @user_agent.user_agent_type = UserAgentType['crawl'] if crawler_detect.is_crawler?
         end
 
         browser = ::Browser.new(user_agent)
+        land_browser = Land::Browser[browser.name]
+        update_browser_color_preferences(land_browser)
 
-        @user_agent.update(
-          browser: Browser[browser.name],
-          device: Device[browser.device.name],
-          platform: Platform[browser.platform.name],
-          browser_version: browser.version
-        )
+        @user_agent.browser = land_browser
+        @user_agent.device = Device[browser.device.name]
+        @user_agent.platform = Platform[browser.platform.name]
+        @user_agent.browser_version = browser.version
+        @user_agent.device_resolution = device_resolution
+
+        begin
+          @user_agent.save! if @user_agent.changed?
+        rescue Standard::Error => e
+          add_error_tag_to_land_span(e)
+        end
 
         @user_agent
       end
+
+      def device_resolution
+        return unless device_width && device_height
+
+        resolution = DeviceResolution.find_or_initialize_by(
+          width: device_width,
+          height: device_height,
+          orientation: device_orientation
+        )
+
+        resolution.device_resolution = "#{device_width}x#{device_height}"
+
+        resolution&.save! if resolution&.changed?
+        resolution
+      rescue ActiveRecord::RecordNotUnique
+        retry
+      rescue ActiveRecord::RecordInvalid => e
+        retry if e.message == 'Validation failed: DeviceResolution has already been taken'
+        add_error_tag_to_land_span(e)
+        raise e
+      end
+
+      def update_browser_color_preferences(browser)
+        return unless browser
+
+        browser.dark_mode = dark_mode
+        browser.light_mode = light_mode
+        browser.no_preference = no_preference
+
+        begin
+          browser.save! if browser.changed?
+        rescue StandardError => e
+          add_error_tag_to_land_span(e)
+        end
```

Comment:
This can also be eliminated
```suggestion
        end
```